src/api/helpers.jl

```diff
@@ -285,6 +285,30 @@ function h5f_get_free_sections(file_id, type, sect_info::AbstractVector{H5F_sect
     return @view(sect_info[1:nsects])
 end
 
+function h5p_get_file_locking(fapl)
+    use_file_locking = Ref{API.hbool_t}(0)
+    ignore_when_disabled = Ref{API.hbool_t}(0)
+    h5p_get_file_locking(fapl, use_file_locking, ignore_when_disabled)
+    return (use_file_locking     = Bool(use_file_locking[]),
+            ignore_when_disabled = Bool(ignore_when_disabled[]))
+end
+
+# Check to see if h5p_set_file_locking should exist
+const _has_h5p_set_file_locking = _has_symbol(:H5Pset_file_locking)
+function has_h5p_set_file_locking()
+   return _has_h5p_set_file_locking
+   #=
+   h5_version = h5_get_libversion()
+   if (h5_version >= v"1.10" && h5_version < v"1.10.7") ||
+      (h5_version >= v"1.12" && h5_version < v"1.12.1") ||
+      (h5_version < v"1.10")
+      return false
+   else
+      return true
+   end
```

Comment:
The commented out lines are an alternative way to check this, but you opted for the `_has_symbol` approach instead?

Reply:
Yes. We could do it either by version or looking up the symbol. Symbols lookup seems more direct, but the version based mechanism may still be informative.